File: 35-search-insert-position/35-search-insert-position.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        
    return binarySearch(nums,target,0,nums.length -1 );
    }
    
    public int binarySearch(int[] nums,int target,int start,int end)
    {
        if (start<=end)
        {
            if (nums[start+(end-start)/2]==target)
            {
                return start+(end-start)/2;
            }
            else if (nums[start+(end-start)/2]<target)
            {
                return binarySearch(nums,target,(start+(end-start)/2)+1,end);
            }
            else{
                return binarySearch(nums,target,0,(start+(end-start)/2)-1);
            }
            
        }
        else {return start;}
        
    }
}

File: 35-search-insert-position/35-search-insert-position.py

                

class Solution:
    def searchInsert(self, nums: List[int], target: int) -> int:
              return self.search(nums, 0, len(nums)-1, target)      

    def search(self, a, s, e, val):
        if s>e:
            return s
        else:
            if val==a[(s+e)//2]:
                return (s+e)//2
            elif val>a[(s+e)//2]:
                return self.search(a, (s+e)//2+1, e, val)
            else:
                return self.search(a, s, (s+e)//2-1, val)
        
        
        
        
        
        
        # iterative approach
#         start , end =  0 , len(nums)-1

#         while start <= end:
#             mid  = (start+end)//2
            
#             if nums[mid] == target:
#                 return mid
#             elif nums[mid] < target:
#                 start = mid+1
#             else:
#                 end = mid-1 
                
#         return start     
        
        
        
        
        
        